src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/HomeOfficeCaseNotificationsHandler.java

```diff
@@ -60,4 +69,18 @@ public boolean canHandle(
 
         return new PreSubmitCallbackResponse<>(asylumCaseWithHomeOfficeData);
     }
+
+    protected Optional<Direction> getLatestNonStandardRespondentDirection(AsylumCase asylumCase) {
+
+        Optional<List<IdValue<Direction>>> maybeExistingDirections = asylumCase.read(AsylumCaseFieldDefinition.DIRECTIONS);
+
+        return maybeExistingDirections
+            .orElseThrow(() -> new IllegalStateException("directions not present"))
+            .stream()
+            .filter(idValue -> idValue.getValue().getTag().equals(DirectionTag.NONE))
+            .filter(idValue -> idValue.getValue().getParties().equals(Parties.RESPONDENT))
+            .max(Comparator.comparing(IdValue::getId))
```

Comment:
@Lee365 I think this could be an issue here. You are filtering first and then picking the max index of the filtered list. But I think that doesn't guarantee that is the latest one. Though it is not a day to day scenario I think it could fail for an edge case. I think they need to be reversed. Could you please think about it and see if it makes sense? We can discuss about that if needed.

Reply:
Corrected the filter. Good point